stdlib/2.7/__builtin__.pyi

```diff
@@ -716,11 +806,11 @@ def next(i: Iterator[_T]) -> _T: ...
 def next(i: Iterator[_T], default: _T) -> _T: ...
 def oct(i: int) -> str: ...  # TODO __index__
 @overload
-def open(file: str, mode: str = 'r', buffering: int = ...) -> BinaryIO: ...
+def open(file: str, mode: str = 'r', buffering: int = ...) -> IO[str]: ...
```

Comment:
Shouldn't it return `IO[bytes]` in PY2? As you've mentioned, the data in PY2 is always 8-bit strings, so `open()` clearly behaves way too different in PY2 and PY3 except for `open(..., '...b')` which is `bytes` in PY2+3. There are more compatible functions `codecs.open()` and `io.open()` where you can return `bytes` or `Text` depending on the mode parameter for both PY2+3.